packages/webapp/src/containers/Filter/Transactions/index.tsx

```diff
@@ -1,5 +1,5 @@
 /*
```

Comment:
The flow for transactions filter is

1. Finances/Report -AND- Finances/TransactionFilter
2. (This component) TransactionsFilterContent
3. FilterGroup + subfilters

Dataflow differs in that:
- Filters are only coupled bi-directionally to the Redux state in Finances/TransactionFilter. The report filters set local component state (via their passed onChange) and have a read-only relationship with the Redux state
- The Redux filter state shape is transformed into the component-used shape here in a common component (this is a good pattern I think)
- The Report Filter state is controlled via the passed onChange, which required modifying the callback passed from FilterGroup to the individual filters in #2942.
  - This is a very clever approach but was hard to type 😄 
  - It means that depending on the calling container, the parameter passed to the onChange by the component is either completely ignored, or is used to the set application state and needs to be a particular shape. It happens to be the wrong shape in the MultiSelect as written, but that combination of container + component is not yet seen in app (it will be)

Reply:
Thank you for these notes, very helpful! I think the structure itself is not entirely different (although the file names change), in the sense that the `TransactionFilter` container defines the ref and relationship with the Redux store, similarly to what the other filter containers do, and the `TransactionsFilterContent` component replaces `FilterPage` in defining the layout and calling `FilterGroup`. `FilterPage` could not be reused because in Finances (and in general from now on) we're not doing separate pages for filters, so I think the Finances pattern will likely need to be repeated in that sense for Animals too. The difference with Animals I think will be that we'll need to filter from the filters modal itself and from the KPIs, but both will need bidirectional flow to the Redux store. The KPIs, unlike the Finances report, will need to dispatch the actual action to Redux to modify what's being displayed. If we do at any point need a report for Animals too though, we'd be well setup to implement the same pattern as here in Finances. 

Re this bit

> It happens to be the wrong shape in the MultiSelect as written, but that combination of container + component is not yet seen in app (it will be)

I think I knew this at the time and chose to ignore any other filter types since like you mention they weren't used in Finances or any similar combination that required this `onChange` function to work properly. Will need to be adjusted, although that entire `MultiSelect` component would now be replaced by the new one anyway.

Reply:
> TransactionsFilterContent component replaces FilterPage in defining the layout and calling FilterGroup

Ah sorry, I should have been more clear -- by layout I meant not of the filter components, but of the overall structure (specifically the title, apply button and the state that determines when that button can be clicked), which in finances were lifted up one level to the container and in the others, sent down one level to the Page.

> FilterPage could not be reused because in Finances (and in general from now on) we're not doing separate pages for filters

Okay now I'm realizing the two situations are even more confusing than I initially thought because... yes, this is exactly right! Aaaaand... all those other filters ARE already modals, not pages! Complete with Drawer behaviour. And they do it by situating the filter container within a Drawer, to which they don't pass buttons, because they let that interior page handle the buttons 😅 

So I think both patterns are still technically viable, but I like the non-nesting approach better for sure; `<Layout />` doesn't look that attractive within a `<Drawer />` and there is less flexibility to style or even change the apply button behaviour.